### PR TITLE
Patching the Gaps: Essential Kibana Configurations added

### PR DESCRIPTION
**Pull request by muhammeddanish**
### Description of the change

This pull request integrates five essential Kibana configurations into our Docker image, enhancing functionality and performance:
1. [server.publicBaseUrl](https://www.elastic.co/guide/en/kibana/8.11/settings.html#server-publicBaseUrl)
2. [xpack.security.encryptionKey](https://www.elastic.co/guide/en/kibana/current/using-kibana-with-security.html#security-configure-settings)
3. [xpack.reporting.encryptionKey](https://www.elastic.co/guide/en/kibana/8.11/reporting-settings-kb.html#encryption-keys)
4. [elasticsearch.requestTimeout](https://www.elastic.co/guide/en/kibana/8.11/settings.html#elasticsearch-requestTimeout)
5. [newsfeed.enabled](https://www.elastic.co/guide/en/kibana/8.11/settings.html)

### Benefits

This PR unlocks further Kibana customization possibilities by adding five essential configurations.
### Possible drawbacks

Without this update, Kibana in versions 7 and above won't work properly with custom domain names (DNS) due to the missing server.publicBaseUrl setting. Other changes enhance its overall functionality.

### Applicable issues

By adding five key configurations, this PR opens doors to enhanced Kibana customization and fixes the issue of DNS access outlined in https://stackoverflow.com/questions/66634552/cant-access-kibana-with-domain.

DNS access for Kibana is now enabled by adding the server.publicBaseUrl configuration, a must for versions 7.x and above.

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution. Please note the version bump from `8.11.3` to `8.11.4` since this version is the upstream Kibana version which at this moment, the latest one is `8.11.3`, see https://github.com/elastic/kibana/releases.

Could you please revert any change related to this version bump?

**Comment by muhammeddanish** [comment]
> Thanks for your contribution. Please note the version bump from `8.11.3` to `8.11.4` since this version is the upstream Kibana version which at this moment, the latest one is `8.11.3`, see https://github.com/elastic/kibana/releases.
> 
> Could you please revert any change related to this version bump?

Thanks @carrodher for the prompt response, I am not sure of how the versioning works in bitnami container images :) 

As recommended, I have removed the modified bump.

File: bitnami/kibana/8/debian-11/rootfs/opt/bitnami/scripts/kibana-env.sh [hunk 1]
```diff
@@ -127,6 +132,16 @@ export KIBANA_FORCE_INITSCRIPTS="${KIBANA_FORCE_INITSCRIPTS:-no}"
...
+export KIBANA_SERVER_PUBLICBASEURL="${KIBANA_SERVER_PUBLICBASEURL:-}"
+export SERVER_PUBLICBASEURL="$KIBANA_SERVER_PUBLICBASEURL"
+export KIBANA_XPACK_SECURITY_ENCRYPTIONKEY="${KIBANA_XPACK_SECURITY_ENCRYPTIONKEY:-}"
+export XPACK_SECURITY_ENCRYPTIONKEY="$KIBANA_XPACK_SECURITY_ENCRYPTIONKEY"
+export KIBANA_XPACK_REPORTING_ENCRYPTIONKEY="${KIBANA_XPACK_REPORTING_ENCRYPTIONKEY:-}"
+export XPACK_REPORTING_ENCRYPTIONKEY="$KIBANA_XPACK_REPORTING_ENCRYPTIONKEY"
+export KIBANA_NEWSFEED_ENABLED="${KIBANA_NEWSFEED_ENABLED:-true}"
+export NEWSFEED_ENABLED="$KIBANA_NEWSFEED_ENABLED"
+export KIBANA_ELASTICSEARCH_REQUESTTIMEOUT="${KIBANA_ELASTICSEARCH_REQUESTTIMEOUT:-30000}"
+export ELASTICSEARCH_REQUESTTIMEOUT="$KIBANA_ELASTICSEARCH_REQUESTTIMEOUT"
```
Comment (dgomezleon) [inline bitnami/kibana/8/debian-11/rootfs/opt/bitnami/scripts/kibana-env.sh:144]: From my understanding, these variables are only for Kibana, so you only need to add this:

```
export KIBANA_SERVER_PUBLICBASEURL="${KIBANA_SERVER_PUBLICBASEURL:-}"
export KIBANA_XPACK_SECURITY_ENCRYPTIONKEY="${KIBANA_XPACK_SECURITY_ENCRYPTIONKEY:-}"
export KIBANA_XPACK_REPORTING_ENCRYPTIONKEY="${KIBANA_XPACK_REPORTING_ENCRYPTIONKEY:-}"
export KIBANA_NEWSFEED_ENABLED="${KIBANA_NEWSFEED_ENABLED:-true}"
export KIBANA_ELASTICSEARCH_REQUESTTIMEOUT="${KIBANA_ELASTICSEARCH_REQUESTTIMEOUT:-30000}"
```

File: bitnami/kibana/8/debian-11/rootfs/opt/bitnami/scripts/libkibana.sh [hunk 2]
```diff
@@ -215,6 +215,31 @@ kibana_initialize() {
...
+        #Controls whether to enable the newsfeed system for the Kibana UI notification center
+        if ! is_boolean_yes "$NEWSFEED_ENABLED"; then
+            kibana_conf_set "newsfeed.enabled" "false" "bool"
+        fi   
+
+        #Time in milliseconds to wait for responses from the back end or Elasticsearch. This value must be a positive integer. Default: 30000
+        if [[ "$ELASTICSEARCH_REQUESTTIMEOUT" != "30000" ]]; then
+            kibana_conf_set "elasticsearch.requestTimeout" "$ELASTICSEARCH_REQUESTTIMEOUT"
+        fi                              
+
```
Comment (dgomezleon) [inline bitnami/kibana/8/debian-11/rootfs/opt/bitnami/scripts/libkibana.sh:242]: All this block applies only for Kibana, so I would suggest this change:

```
        # Kibana override configuration
        if [[ "$SERVER_FLAVOR" = "kibana" ]]; then
            if is_boolean_yes "$KIBANA_DISABLE_STRICT_CSP"; then
                kibana_conf_set "csp.strict" "false" "bool"
            fi
            if ! is_empty_value "$KIBANA_SERVER_PUBLICBASEURL"; then
                kibana_conf_set "server.publicBaseUrl" "$KIBANA_SERVER_PUBLICBASEURL"
            fi
            if ! is_empty_value "$KIBANA_XPACK_SECURITY_ENCRYPTIONKEY"; then
                kibana_conf_set "xpack.security.encryptionKey" "$KIBANA_XPACK_SECURITY_ENCRYPTIONKEY"
            fi
            if ! is_empty_value "$KIBANA_XPACK_REPORTING_ENCRYPTIONKEY"; then
                kibana_conf_set "xpack.reporting.encryptionKey" "$KIBANA_XPACK_REPORTING_ENCRYPTIONKEY"
            fi
            if ! is_boolean_yes "$KIBANA_NEWSFEED_ENABLED"; then
                kibana_conf_set "newsfeed.enabled" "false" "bool"
            fi
            if [[ "$KIBANA_ELASTICSEARCH_REQUESTTIMEOUT" != "30000" ]]; then
                kibana_conf_set "elasticsearch.requestTimeout" "$KIBANA_ELASTICSEARCH_REQUESTTIMEOUT"
            fi
        fi
```

**Review by dgomezleon — CHANGES_REQUESTED** [review]

**Comment by dgomezleon** [comment]
Hi @muhammeddanish,

Thanks for contributing! 
I have added some comments to the PR. Also, please note this should also be applied to branch 7.

**Comment by muhammeddanish** [comment]
> Hi @muhammeddanish,
> 
> Thanks for contributing! I have added some comments to the PR. Also, please note this should also be applied to branch 7.

Thanks @dgomezleon !!
I have updated the code with the suggested changes.

**Review by dgomezleon — APPROVED** [review]
LGTM

**Comment by muhammeddanish** [comment]
> LGTM

Thank you :)  

Could you please help me to merge this PR?

**Pull request merged**